pkg/pulumiyaml/run.go

```diff
@@ -265,6 +243,9 @@ func LoadYAMLBytes(filename string, source []byte) (*ast.TemplateDecl, syntax.Di
 	if t.Configuration.Entries != nil {
 		diags = append(diags, syntax.Warning(nil, "Pulumi.yaml: root-level `configuration` field is deprecated; please use `config` instead.", ""))
 	}
+	if !isComponent && t.Namespace != nil {
+		diags = append(diags, syntax.Error(nil, "Pulumi.yaml: root-level `namespace` field is only supported in components.", ""))
```

Comment:
Wouldn't the right way to frame this as only supported in plugins (PulumiPlugin.yaml not Pulumi.yaml)?

Reply:
Maybe component plugins?

Reply:
> components could be local in Pulumi.yaml

No they can't be currently. We explicitly disallow that: https://github.com/pulumi/pulumi-yaml/blob/master/pkg/pulumiyaml/run.go#L478.  It's something we could introduce later though, so I'll rename it to `component plugins`.

Reply:
Hrm linking to that line makes me thing we're probably better off doing that check there, then we don't need to pass around `isComponent` everywhere either.